### PR TITLE
feat!: establish baseline Node.js version support

### PR DESCRIPTION
**Pull request by boneskull**
I wonder if adding the `engines` field is going to break somebody.  I guess I'd only really care if their package manager's _default behavior_ is like enabling `engineStrict`?  Is that a thing?

### CODE REVIEW

**Comment by boneskull** [comment]
@nzakas Given this package is not doing anything especially fancy, would you consider dropping macOS from the build matrix?  I'm not sure it's buying us much.

**Comment by nzakas** [comment]
No objections from me.

**Comment by boneskull** [comment]
@nzakas I've created #186, but _this_ PR still needs an approval. 😄

**Comment by boneskull** [comment]
@nzakas *friendly ping for approval*

**Comment by nzakas** [comment]
Gah sorry, just been buried lately. Thanks for the followup ping.

File: package.json [hunk 1]
```diff
@@ -70,5 +70,8 @@
     "rollup": "4.23.0",
     "typescript": "5.6.3",
     "yorkie": "2.0.0"
+  },
+  "engines": {
+    "node": "^16.20.0 || ^18.0.0 || ^20.0.0 || ^22.0.0"
```
Comment (nzakas) [inline package.json:75]: I generally consider changing explicitly supported node versions as a breaking change. I don't mind doing it, but I do think we should bump major version for clarity. WDYT?

Comment (nzakas) [reply package.json:75]: It's handled via release-please, so if we update the title of this PR to be `"feat!: Set Node.js support"` (or some such thing with "feat!" at the front), when this is merged, release-please creates a major version bump.

Comment (boneskull) [reply package.json:75]: great, right. I probably should have checked 😄

**Comment by boneskull** [comment]
@nzakas OK, this is ready for final approval.

**Review by nzakas — APPROVED** [review]
LGTM.

**Pull request merged**